### PR TITLE
Add mutation to update content taxonomy

### PR DESCRIPTION
**Pull request by solocommand**
```graphql
mutation UpdateCategories($input: ContentTaxonomyMutationInput!) {
  contentTaxonomy(input: {
    id: 21577383,
    addIds: [3195777, 3197737],
    removeIds: [3195775]
  }) {
    id
    taxonomyIds
  }
}
```
```json
{
  "data": {
    "contentTaxonomy": {
      "id": 21577383,
      "taxonomyIds": [3195777, 3197737]
    }
  }
}
```

### CODE REVIEW

**Comment by solocommand** [comment]
[Add support for selecting taxonomy categories](https://trello.com/c/IEWOOTYV/130-add-support-for-selecting-taxonomy-categories)

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 1]
```diff
@@ -1888,6 +1888,39 @@ module.exports = {
...
+
+      // Validate taxonomy ids
+      const ids = [...addIds, ...removeIds, ...getAsArray(doc, 'taxonomy').map(r => r.oid)];
+      const query = { status: 1, _id: { $in: ids } };
+      const docs = await basedb.find('platform.Taxonomy', query, { projection: { type: 1 } });
+      const taxonomyMap = docs.reduce((map, term) => {
+        map.set(term._id, `platform/content/${dasherize(term.type)}`);
+        return map;
+      }, new Map());
+
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js:1913]: ```removeIds.forEach((id) => taxomomyMap.delete(id));```

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 2]
```diff
@@ -1888,6 +1888,39 @@ module.exports = {
...
+      // 2.0rcpi is hardcoded to use the `set` action, so we update all instead of assign/remove.
+      const taxonomy = getAsArray(doc, 'taxonomy')
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: Remove this

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 3]
```diff
@@ -1888,6 +1888,39 @@ module.exports = {
...
+        .map(t => t.oid)
+        .filter(tid => !removeIds.includes(tid));
+      addIds.forEach(tid => taxonomy.push(tid));
+
+      const body = new Base4RestPayload({ type });
+      body.set('id', id);
+      body.setLinks('taxonomy', taxonomy.map(tid => ({ id: tid, type: taxonomyMap.get(tid) })));
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: `[...taxonomyMap].map(([id, type]) => ({ id, type }))`

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

**Review by zarathustra323 — APPROVED** [review]

**Pull request merged**